src/coreclr/jit/lower.cpp

```diff
@@ -5268,7 +5272,7 @@ bool Lowering::LowerUnsignedDivOrMod(GenTreeOp* divMod)
             adjustedDividend->SetRegNum(REG_RAX);
 #endif
 
-        divisor->gtType = TYP_I_IMPL;
+        divisor->gtType = simpleMul ? TYP_INT : TYP_I_IMPL;
```

Comment:
This will only happen for this ARM64 case, right? The Jit doesn't expect `TYP_INT`/`TYP_LONG` mixes in binary operands.

Reply:
Yes, this should be under `TARGET_ARM64`.